### PR TITLE
Create time lapse video (visualize sub package)

### PR DESCRIPTION
**Pull request by DannieSheng**
**Describe your changes**

1. Add a function to create time-lapse video in the `visualize` sub-package 
2. Create a new module that prints out warning messages

This PR was updated so the reading of the image data set can be done outside of the function using the io module.
Image resizing was also removed since it only zero-padded or cropped the images.

**Type of update**
Is this a:
* New feature

Additional context

### CODE REVIEW

File: plantcv/plantcv/visualize/time_lapse_video.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
+# Create time-lapse videos with input directory of images
+
+import os
+import cv2
+import numpy as np
+import copy
```
Comment (JorgeGtz) [inline plantcv/plantcv/visualize/time_lapse_video.py:6]: not used

File: plantcv/plantcv/visualize/time_lapse_video.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+import mimetypes
+from plantcv.plantcv import params
```
Comment (JorgeGtz) [inline plantcv/plantcv/visualize/time_lapse_video.py:8]: not used

File: plantcv/plantcv/visualize/time_lapse_video.py [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+            list_r.append(img.shape[0])
+            list_c.append(img.shape[1])
+            imgs.append(img)
+        max_c, max_r = np.max(list_c), np.max(list_r)
+
+        # If the frame size is not provided, use the largest size of the images as the frame size
+        size_frame = size_frame or (max_c, max_r)
+
+        if not (len(np.unique(list_r)) == 1 and len(np.unique(list_c)) == 1):
+            warnings.warn(f"The sizes of images are not the same, an image resizing (cropping or zero-padding) will be done "
```
Comment (JorgeGtz) [inline plantcv/plantcv/visualize/time_lapse_video.py:107]: Is warnings.warn working as expected? Noah mentioned it only prints the warning in developing mode or something like that

**Review by JorgeGtz — CHANGES_REQUESTED** [review]
Looks good! I made a few comments in time_lapse_video.py, we can discuss them before merging. An overall comment is that some lines are too long, Noah recommends breaking them at 120 characters.

File: plantcv/plantcv/visualize/time_lapse_video.py [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+    ## Get the list of image files in the given directory and sort them alphabetically by their names
+    temp_list = []
+    for f in os.listdir(img_directory):
+        type_f = mimetypes.guess_type(f)[0]
+        if type_f:
+            if type_f.startswith('image'):
+                temp_list.append(f)
+    if len(temp_list) <= 0:
+        fatal_error("There is no file in the provided folder that is an image, please check the provided directory!")
+    else:
```
[the remark this answers is not in the document]
Comment (DannieSheng) [reply plantcv/plantcv/visualize/time_lapse_video.py:58]: I've fixed it

File: plantcv/plantcv/visualize/time_lapse_video.py [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+            list_img_no_ext = [os.path.splitext(f)[0] if os.path.splitext(f)[1] else f for f in list_img]
+            list_img_ = []
+            # check if the images in the given list exist in the directory by comparing both lists
+            for (f,f_) in zip(list_img_no_ext,list_img):
+                try:
+                    temp_list_no_ext.index(f)
+                    list_img_.append(f_)
+                except ValueError:
+                    continue
+            list_img = list_img_
```
[the remark this answers is not in the document]
Comment (DannieSheng) [reply plantcv/plantcv/visualize/time_lapse_video.py:75]: I don't have a good idea of how to do that. I just tried using set intersection to find the list of "extension free" file names that is provided by user and is in the provided directory. However with this "extension free" list of image names, we still need the extensions for every file. The assumption is that the file extensions are not necessarily the same (though normally it should not happen :)). Do you have good suggestions of this?

[the remark this answers is not in the document]
Comment (nfahlgren) [reply plantcv/plantcv/visualize/time_lapse_video.py:75]: I haven't used this Python feature yet, but I think @JorgeGtz is suggesting you could make `temp_list_no_ext` and `list_img_no_ext` Python sets and use the `intersection` method to find the union of the two: https://www.w3schools.com/python/ref_set_intersection.asp

[the remark this answers is not in the document]
Comment (DannieSheng) [reply plantcv/plantcv/visualize/time_lapse_video.py:75]: Yes, I've come to that step. The challenge is that after creating the `final_list_no_ext`, we will need the extension for every file. And the assumption is that not every file has the same file extension, i.e. the file extension is "file dependent". Although in most cases every image file should have the same extension. So there might be an extra step of finding the file extension for every file. But there might be solutions that I did not come up with.

[the remark this answers is not in the document]
Comment (JorgeGtz) [reply plantcv/plantcv/visualize/time_lapse_video.py:75]: What about something like this:
```
temp_set = set(temp_list_no_ext)
for idx, f in enumerate(list_img_no_ext):
    if f in temp_set:
        list_img_.append(list_img[idx])
```

[the remark this answers is not in the document]
Comment (DannieSheng) [reply plantcv/plantcv/visualize/time_lapse_video.py:75]: That's a good idea! Thank you!

**Review by JorgeGtz — APPROVED** [review]

**Review by HaleySchuhl — DISMISSED** [review]

File: plantcv/plantcv/visualize/time_lapse_video.py [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+import numpy as np
+import mimetypes
+from plantcv.plantcv import params
+from plantcv.plantcv import fatal_error
+from plantcv.plantcv.transform import resize
+# import warnings
+from plantcv.plantcv import warn
+
+
+def time_lapse_video(img_directory, list_img=None, auto_sort = True, suffix_img=None, size_frame=None, fps=29.97,
```
Comment (nfahlgren) [inline plantcv/plantcv/visualize/time_lapse_video.py:14]:
```suggestion
def time_lapse_video(img_directory, list_img=None, auto_sort=True, suffix_img=None, size_frame=None, fps=29.97,
```

**Review by nfahlgren — CHANGES_REQUESTED** [review]

**Comment by nfahlgren** [comment]
Some other changes to consider as I was using this yesterday:

* In debug mode all the frames are plotted, which can open too many plots for Jupyter - we need a different visualization (or none)
* The output filename is created from the input name and path, these can just be a single option for an output filename (which can include a path)

**Review by HaleySchuhl — APPROVED** [review]

**Review by HaleySchuhl — APPROVED** [review]

**Pull request merged**